### ISSUE TITLE
T::Struct should error if a property name is duplicated

### ISSUE THREAD

**Issue by Morriar** [sorbet/sorbet#3802]
#### Input

[→ View on sorbet.run](https://sorbet.run/#%23%20typed%3A%20true%0A%0Aclass%20Foo%20%3C%20T%3A%3AStruct%0A%20%20prop%20%3Afoo%2C%20String%0A%20%20const%20%3Afoo%2C%20String%0Aend%0A%0AFoo.new(foo%3A%20%22foo%22))

```ruby
# typed: true

class Foo < T::Struct
  prop :foo, String
  const :foo, String
end

Foo.new(foo: "foo")
```

#### Observed output

```
editor.rb:8: Missing required keyword argument foo for method Foo#initialize https://srb.help/7004
     8 |Foo.new(foo: "foo")
        ^^^^^^^^^^^^^^^^^^^
Errors: 1
```

#### Expected behavior

I was expecting an error directly in the struct:

```
class Foo < T::Struct
  prop :foo, String
  const :foo, String # error: Struct field `foo` already defined
end
```

**Comment by jez** [comment]
For what it's worth, this is a runtime error already:

```ruby
❯ ruby foo.rb
Traceback (most recent call last):
        10: from foo.rb:4:in `<main>'
         9: from foo.rb:6:in `<class:Foo>'
         8: from /home/jez/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/sorbet-runtime-0.5.6052/lib/types/private/methods/_methods.rb:228:in `block in _on_method_added'
         7: from /home/jez/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/sorbet-runtime-0.5.6052/lib/types/private/methods/call_validation.rb:126:in `validate_call'
         6: from /home/jez/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/sorbet-runtime-0.5.6052/lib/types/private/methods/call_validation.rb:126:in `call'
         5: from /home/jez/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/sorbet-runtime-0.5.6052/lib/types/props/_props.rb:136:in `const'
         4: from /home/jez/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/sorbet-runtime-0.5.6052/lib/types/props/_props.rb:105:in `prop'
         3: from /home/jez/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/sorbet-runtime-0.5.6052/lib/types/props/decorator.rb:354:in `prop_defined'
         2: from /home/jez/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/sorbet-runtime-0.5.6052/lib/types/props/serializable.rb:202:in `add_prop_definition'
         1: from /home/jez/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/sorbet-runtime-0.5.6052/lib/types/props/optional.rb:61:in `add_prop_definition'
/home/jez/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/sorbet-runtime-0.5.6052/lib/types/props/decorator.rb:49:in `add_prop_definition': Attempted to redefine prop :foo that's already defined without specifying :override => true: {:type=>String, :type_object=>#<T::Types::Simple:0x00000000020006c0 @raw_type=String, @name="String">, :accessor_key=>:@foo, :sensitivity=>nil, :pii=>nil, :extra=>nil, :setter_proc=>#<Proc:0x0000000002179628@/home/jez/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/sorbet-runtime-0.5.6052/lib/types/props/private/setter_factory.rb:54>, :serialized_form=>"foo", :fully_optional=>false, :need_nil_read_check=>nil} (ArgumentError)
```

so it looks like this is a request for a static error.

**Comment by jez** [comment]
Fixed in #8276
